packages/tx/src/eip2930Transaction.ts

```diff
@@ -278,9 +278,11 @@ export default class AccessListEIP2930Transaction extends BaseTransaction<Access
   /**
    * Returns the serialized unsigned tx (hashed or raw), which is used to sign the transaction.
    *
-   * @param hashMessage - Return hashed message if set to true
+   * @param hashMessage - Return hashed message if set to true (default: true)
    */
-  getMessageToSign(hashMessage = true) {
+  getMessageToSign(hashMessage: false): Buffer[]
+  getMessageToSign(hashMessage?: true): Buffer
+  getMessageToSign(hashMessage = true): Buffer | Buffer[] {
```

Comment:
I am still getting familiar with function overloading in TypeScript to better infer the returned type, so I'm a bit perplexed why I needed to add this final `: Buffer | Buffer[]` when the same exact line in [`legacyTransaction.ts:156`](https://github.com/ethereumjs/ethereumjs-monorepo/pull/1214/commits/c1f14fed9751bdb495d508938331b038ba567e4c#diff-739632585d309f4741161671245094379f440cf7ed2d8047dc24cd16ced6e712R156) was fine without it (error was: `This overload signature is not compatible with its implementation signature.`)